packages/voting/src/deposit.rs

```diff
@@ -27,20 +28,28 @@ pub struct DepositInfo {
     /// proposal.
     pub deposit: Uint128,
     /// If failed proposals should have their deposits refunded.
-    pub refund_failed_proposals: bool,
+    pub refund_policy: DepositRefundPolicy,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum DepositRefundPolicy {
+    Always,
+    OnlyPassed,
+    Never,
 }
 
 /// Counterpart to the `DepositInfo` struct which has been processed.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct CheckedDepositInfo {
     /// The address of the cw20 token to be used for proposal
     /// deposits.
-    pub token: Addr,
+    pub token: AssetInfo,
     /// The number of tokens that must be deposited to create a
     /// proposal.
     pub deposit: Uint128,
     /// If failed proposals should have their deposits refunded.
-    pub refund_failed_proposals: bool,
+    pub refund_policy: DepositRefundPolicy,
 }
```

Comment:
This is a subtle one, but this change actually breaks v1 -> v2 migration logic. That logic expects that a v1 deposit info struct will deserialize into this type. By changing the layout, we break that.

Can probably hold off on updating the code until I've got a pre-propose module PR up and we decide if we'd like to merge this.